plugins/module_utils/network/nxos/facts/legacy/base.py

```diff
@@ -270,13 +270,16 @@ def populate(self):
                 self.populate_ipv6_interfaces(interfaces)
 
         data = self.run("show lldp neighbors", output="json")
-        if data:
+
+        try:
             if isinstance(data, dict):
                 self.facts["neighbors"].update(
                     self.populate_structured_neighbors_lldp(data),
                 )
             else:
                 self.facts["neighbors"].update(self.populate_neighbors(data))
+        except Exception as e:
```

Comment:
This might mask actual errors. We need to narrow the scope of the exception class being used and move the try-except closer to the source of potential exception generating code.

Reply:
IMO, the try-except needs to be moved closer to where the `KeyError` can originate from.